js/models/Settings.js

```diff
@@ -16,6 +16,8 @@ export default class extends BaseModel {
       refundPolicy: '',
       blockedNodes: [],
       storeModerators: [],
+      shippingAddresses: new ShippingAddresses(),
```

Comment:
This is the new set-up to differentiate between an optional nested model and a required one... in either case, you are still declaring them in the nested getter:

```
get nested() {
  return {
    nestedModelRequired: Model,
    nestedModelOptional: Model,
  }
}
```

The difference is that the required model should be instantiated in the default method:

```
defaults() {
  return {
    nestedModelRequired: new Model(),
  }
}
```

Basically, what this structure does is that when the parent model is created, any nest model instantiated in the defaults will also be created, whereas "optional" models will be created the first time they are set.